src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs

```diff
@@ -1036,29 +1036,11 @@ private void Grow(int requiredSize)
 
             Debug.Assert(BytesPending != 0);
 
-            _memory = default;
```

Comment:
Removed this because _memory is reset in each branch below and if `GetMemory(...)` fails for some reason (say overflow), subsequent writes were going down the `FirstCallToGetMemory` path and failing on the BytesPending != 0 `Debug.Assert`.

This should be side effect free, at least in the common/happy cases. I don't know if we should still set it to default to make the GC happy in the exception path (no more references to the underlying array).
https://github.com/dotnet/corefx/blob/c707378f2a8ae1da5ebc5d02bbe2e0a3fd26a6a0/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs#L1082-L1085